src/service/src/secrets.rs

```diff
@@ -11,53 +11,121 @@ use std::path::PathBuf;
 use std::sync::Arc;
 
 use clap::ArgEnum;
+use mz_aws_secrets_controller::AwsSecretsClient;
 use mz_orchestrator_kubernetes::secrets::KubernetesSecretsReader;
 use mz_orchestrator_process::secrets::ProcessSecretsReader;
 use mz_secrets::SecretsReader;
 
-#[derive(clap::Parser)]
+#[derive(clap::Parser, Clone, Debug)]
 pub struct SecretsReaderCliArgs {
     /// The secrets reader implementation to use.
     #[structopt(long, arg_enum, env = "SECRETS_READER")]
-    secrets_reader: SecretsReaderKind,
+    pub secrets_reader: SecretsControllerKind,
     /// When using the process secrets reader, the directory on the filesystem
     /// where secrets are stored.
     #[structopt(
         long,
-        required_if_eq("secrets-reader", "process"),
-        env = "SECRETS_READER_PROCESS_DIR"
+        required_if_eq("secrets-reader", "local-file"),
+        env = "SECRETS_READER_LOCAL_FILE_DIR"
     )]
-    secrets_reader_process_dir: Option<PathBuf>,
+    pub secrets_reader_local_file_dir: Option<PathBuf>,
     /// When using the Kubernetes secrets reader, the Kubernetes context to
     /// load.
     #[structopt(
         long,
         required_if_eq("secrets-reader", "kubernetes"),
         env = "SECRETS_READER_KUBERNETES_CONTEXT"
     )]
-    secrets_reader_kubernetes_context: Option<String>,
+    pub secrets_reader_kubernetes_context: Option<String>,
+    /// When using the AWS secrets reader, we need both of the following.
+    #[structopt(
+        long,
+        required_if_eq("secrets-reader", "aws-secrets-manager"),
+        env = "SECRETS_READER_AWS_REGION"
+    )]
+    pub secrets_reader_aws_region: Option<String>,
+    #[structopt(
+        long,
+        required_if_eq("secrets-reader", "aws-secrets-manager"),
+        env = "SECRETS_READER_AWS_PREFIX"
+    )]
+    pub secrets_reader_aws_prefix: Option<String>,
 }
 
-#[derive(ArgEnum, Debug, Clone)]
-enum SecretsReaderKind {
-    Process,
+#[derive(ArgEnum, Debug, Clone, Copy)]
+pub enum SecretsControllerKind {
```

Comment:
This rename doesn’t jive with the name of the other arguments declared here! Fine for now to get the release unblocked though.